plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemColor.java

```diff
@@ -3,156 +3,71 @@
-                    || mat == Material.LINGERING_POTION
-                    || mat == Material.SPLASH_POTION) {
-                PotionMeta pm = (PotionMeta) item.getItemMeta();
-                if (!pm.hasColor()) {
-                    return BukkitColorExtensions.fromColor(Color.WHITE).getObjectAttribute(attribute.fulfill((1)));
-                }
-                return BukkitColorExtensions.fromColor(pm.getColor()).getObjectAttribute(attribute.fulfill((1)));
+        if (getItemMeta() instanceof MapMeta mapMeta) {
+            if (!mapMeta.hasColor() || mapMeta.getColor() == null) {
+                return null;
             }
-            if (mat == Material.FILLED_MAP) {
-                MapMeta mapMeta = (MapMeta) item.getItemMeta();
-                if (!mapMeta.hasColor()) {
-                    return null;
-                }
-                return BukkitColorExtensions.fromColor(mapMeta.getColor()).getObjectAttribute(attribute.fulfill(1));
+            return BukkitColorExtensions.fromColor(mapMeta.getColor());
+        }
+        if (getItemMeta() instanceof PotionMeta potionMeta) {
+            if (!potionMeta.hasColor() || potionMeta.getColor() == null) {
+                return null;
             }
-            return BukkitColorExtensions.fromColor(((LeatherArmorMeta) item.getItemMeta()).getColor()).getObjectAttribute(attribute.fulfill(1));
+            return BukkitColorExtensions.fromColor(potionMeta.getColor());
         }
-
         return null;
     }
 
     @Override
-    public String getPropertyString() {
-        Material mat = item.getBukkitMaterial();
-        if (mat == Material.POTION
-                || mat == Material.LINGERING_POTION
-                || mat == Material.SPLASH_POTION) {
-            PotionMeta pm = (PotionMeta) item.getItemMeta();
-            if (!pm.hasColor()) {
-                return null;
-            }
-            return BukkitColorExtensions.fromColor(pm.getColor()).identify();
+    public void setPropertyValue(ColorTag color, Mechanism mechanism) {
+        if (getItemMeta() instanceof LeatherArmorMeta leatherArmorMeta) {
+            leatherArmorMeta.setColor(BukkitColorExtensions.getColor(color));
+            getItemStack().setItemMeta(leatherArmorMeta);
+            return;
         }
-        if (mat == Material.FILLED_MAP) {
-            MapMeta mapMeta = (MapMeta) item.getItemMeta();
-            if (!mapMeta.hasColor()) {
-                return null;
-            }
-            return BukkitColorExtensions.fromColor(mapMeta.getColor()).identify();
+        if (getItemMeta() instanceof MapMeta mapMeta) {
+            mapMeta.setColor(BukkitColorExtensions.getColor(color));
+            getItemStack().setItemMeta(mapMeta);
+            return;
+        }
+        if (getItemMeta() instanceof PotionMeta potionMeta) {
+            potionMeta.setColor(BukkitColorExtensions.getColor(color));
+            getItemStack().setItemMeta(potionMeta);
         }
```

Comment:
Don't really need an `if` for the last check here, as you can know for sure it's that type (since it isn't any of the other ones) - can probably just have a quick in-line `editMeta` call?